stdlib/Logging/src/Logging.jl

```diff
@@ -29,6 +30,41 @@ for sym in [
     @eval const $sym = Base.CoreLogging.$sym
 end
 
+"""
+    @create_log_macro(name::Symbol, level::Int, color::Union{Int,Symbol})
+
+Creates a custom log macro like `@info`, `@warn` etc. with a given `name`, `level` and
+`color`. The macro created is named with the lowercase form of `name` but the given form
+is used for the printing.
+
+The available color keys can be seen by typing `Base.text_colors` in the help mode of the REPL
+
+```julia-repl
+julia> @create_log_macro(:MyLog, 200, :magenta)
+@mylog (macro with 1 method)
+
+julia> @mylog "hello"
+[ MyLog: hello
+```
+"""
+macro create_log_macro(name, level, color)
+    macro_name = Symbol(lowercase(string(name)))
+    macro_string = QuoteNode(name)
+    loglevel = LogLevel(level)
+    if loglevel in (BelowMinLevel, Debug, Info, Warn, Error, AboveMaxLevel)
+        throw(ArgumentError("Cannot use the same log level as a built in log macro"))
+    end
+    if haskey(custom_log_levels, loglevel)
+        throw(ArgumentError("Custom log macro already exists for given log level"))
+    end
```

Comment:
I guess we're ok with reusing existing colors? Maybe it would be helpful to at least note the colors the builtin log levels are already using?

Reply:
I think it's ok to leave that up to the user. They can see the colors of existing logs and figure out what they want from that I think